backend/src/main/java/com/example/backend/service/GifService.java

```diff
@@ -36,4 +36,19 @@ public Gif editGifById(GifWithoutId g, String id) {
         Gif editedGif = new Gif(gif.getId(), g.getName(), g.getDescription(), g.getPrice());
         return this.gifRepository.save(editedGif);
     }
-}
\ No newline at end of file
+
+
+    public Gif getGifById(String id) {
```

Comment:
are you finding the gif by id in the gifRepository ans then save it there?
can't you simply "return gif"?